sys/net/sock/sock_util.c

```diff
@@ -116,30 +117,35 @@ static char* _find_pathstart(const char *url)
 
 int sock_urlsplit(const char *url, char *hostport, char *urlpath)
 {
+    assert(url);
     char *hoststart = _find_hoststart(url);
     if (!hoststart) {
         return -EINVAL;
     }
 
     char *pathstart = _find_pathstart(hoststart);
 
-    size_t hostlen = pathstart - hoststart;
-    /* hostlen must be smaller SOCK_HOSTPORT_MAXLEN to have space for the null
-     * terminator */
-    if (hostlen > SOCK_HOSTPORT_MAXLEN - 1) {
-        return -EOVERFLOW;
+    if (hostport) {
+        size_t hostlen = pathstart - hoststart;
+        /* hostlen must be smaller SOCK_HOSTPORT_MAXLEN to have space for the null
+        * terminator */
+        if (hostlen > SOCK_HOSTPORT_MAXLEN - 1) {
+            return -EOVERFLOW;
+        }
+        memcpy(hostport, hoststart, hostlen);
+        *(hostport + hostlen) = '\0';
```

Comment:
```suggestion
        hostport[hostlen] = '\0';
```
Note: i know you didn't write that, but considering there are not many chances to do little fixes like this...